packages/common-integration-tests/src/auth-client.ts

```diff
@@ -1340,6 +1250,98 @@ export function runAuthClientTests(
       );
     });
 
+    it('can use disposable tokens to read and write to caches and topics like usual', async () => {
+      const superUserTokenResponse =
+        await sessionTokenAuthClient.generateAuthToken(
+          SUPER_USER_PERMISSIONS,
+          ExpiresIn.seconds(10)
+        );
+      expect(superUserTokenResponse).toBeInstanceOf(GenerateAuthToken.Success);
+
+      const authClient = authTokenAuthClientFactory(
+        (superUserTokenResponse as GenerateAuthToken.Success).authToken
+      );
+
+      const tokenResponse = await authClient.generateDisposableToken(
+        {
+          permissions: [
+            {role: CacheRole.ReadWrite, cache: FGA_CACHE_1},
+            {
+              role: TopicRole.PublishSubscribe,
+              cache: FGA_CACHE_2,
+              topic: AllTopics,
+            },
+          ],
+        },
+        ExpiresIn.seconds(60)
+      );
+      expect(tokenResponse).toBeInstanceOf(GenerateDisposableToken.Success);
+
+      const token = (tokenResponse as GenerateDisposableToken.Success)
+        .authToken;
+      const cacheClient = cacheClientFactory(token);
+      const topicClient = topicClientFactory(token);
+
+      // can get and set items in FGA_CACHE_1 but not FGA_CACHE_2
+      const setResp1 = await cacheClient.set(FGA_CACHE_1, 'cow', 'meow');
+      expect(setResp1).toBeInstanceOf(CacheSet.Success);
+
+      const getKey1 = await cacheClient.get(FGA_CACHE_1, 'cow');
+      expect(getKey1).toBeInstanceOf(CacheGet.Hit);
+      expect(getKey1.value()).toEqual('meow');
+
+      const setResp3 = await cacheClient.set(
+        FGA_CACHE_2,
+        FGA_CACHE_2_KEY,
+        'woof'
+      );
+      expect(setResp3).toBeInstanceOf(CacheSet.Error);
+      const setError1 = setResp3 as CacheSet.Error;
+      expect(setError1.errorCode()).toEqual(MomentoErrorCode.PERMISSION_ERROR);
+      expect(setError1.message()).toContain('Insufficient permissions');
+
+      await expectPermissionDeniedForGet(
+        cacheClient,
+        FGA_CACHE_2,
+        FGA_CACHE_2_KEY
+      );
+
+      // can publish and subscribe topics in FGA_CACHE_2 but not FGA_CACHE_1
+      const subResp1 = await topicClient.subscribe(
+        FGA_CACHE_2,
+        'breaking news!',
+        trivialHandlers
+      );
+      expect(subResp1).toBeInstanceOf(TopicSubscribe.Subscription);
+
+      const pubResp1 = await topicClient.publish(
+        FGA_CACHE_2,
+        'breaking news!',
+        'humans landed on Mars!'
+      );
+      expect(pubResp1).toBeInstanceOf(TopicPublish.Success);
+
+      const subResp2 = await topicClient.subscribe(
+        FGA_CACHE_1,
+        'breaking news!',
+        trivialHandlers
+      );
+      expect(subResp2).toBeInstanceOf(TopicSubscribe.Error);
```

Comment:
I found we can actually [extend the jest matchers](https://jestjs.io/docs/expect#expectextendmatchers) after I wrote the wrapper functions `expectPermissionDeniedForGet` and `expectPermissionDeniedForSet`. If you could try it out for this new test, it will greatly simplify writing more of these tests. No need to refactor existing tests right now, and I understand if you don't have the time.
E.g. we could write this test like
```
expect(getKey1).toBeHit(expectedValue);
expect(setResp).toBePermissionDeniedForSet();
expect(subResp2).toBePermissionDeniedForTopicSubscribe();
```